Incorrect comparison in unversioned.MatchesServerVersion

The following code compares cVer.GitTreeState against itself. The RHS should probably use sVer instead:

```
if cVer.GitVersion != sVer.GitVersion || cVer.GitCommit != sVer.GitCommit || cVer.GitTreeState != cVer.GitTreeState {
	return fmt.Errorf("server version (%#v) differs from client version (%#v)!\n", sVer, cVer)
}
```

```
/src/github.com/kubernetes/kubernetes/pkg/client/unversioned/helper.go:117:79: identical expressions on the left and right side of the '!=' operator
```

--- a/pkg/client/unversioned/helper.go
+++ b/pkg/client/unversioned/helper.go
@@ -143,7 +143,7 @@ func MatchesServerVersion(client *Client, c *restclient.Config) error {
 		return fmt.Errorf("couldn't read version from server: %v\n", err)
 	}
 	// GitVersion includes GitCommit and GitTreeState, but best to be safe?
-	if cVer.GitVersion != sVer.GitVersion || cVer.GitCommit != sVer.GitCommit || cVer.GitTreeState != cVer.GitTreeState {
+	if cVer.GitVersion != sVer.GitVersion || cVer.GitCommit != sVer.GitCommit || cVer.GitTreeState != sVer.GitTreeState {
 		return fmt.Errorf("server version (%#v) differs from client version (%#v)!\n", sVer, cVer)
 	}
 

